src/main/groovy/wooga/gradle/unity/tasks/ProjectManifestTask.groovy

```diff
@@ -0,0 +1,49 @@
+package wooga.gradle.unity.tasks
+
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import org.gradle.process.ExecResult
+import wooga.gradle.unity.UnityTask
+import wooga.gradle.unity.models.ResolutionStrategy
+import wooga.gradle.unity.traits.UnityPackageSpec
+
+abstract class ProjectManifestTask extends DefaultTask implements
+    UnityPackageSpec {
+
+    final static String manifestFileName = "manifest.json"
+    final static String lockFileName = "packages-lock.json"
+
+    abstract void onExecute(Object manifest)
```

Comment:
I would also change the naming scheme here. As I commented on other places. There is no `on...` naming pattern in java/groove/gradle which totally confuses me. I would name it `modifyManifest` or something down the line.

Reply:
In this case, the naming scheme comes from wanting to naturally imply that this method will be called after `execute`.

Reply:
But yes, I will refactor it.